node_exporter.go

```diff
@@ -97,6 +99,9 @@ func main() {
 		log.Infof(" - %s", n)
 	}
 
+	// Use our shared code to run server and exit on error. Upstream's code below will not be executed.
+	exporter_shared.RunServerFunc("Node", *listenAddress, *metricsPath, handler)
```

Comment:
That line calls `handler` function defined at https://github.com/percona/node_exporter/pull/10/commits/3803ddc0e8ca24f1b436717a61add330a175b31b#diff-96274f342a5b55fb60671bb66a2ea228R36. That completely skips basic authentication.